src/mongo/MongoCommands.ts

```diff
@@ -18,16 +18,61 @@ import { MongoFindOneResultEditor } from './editors/MongoFindOneResultEditor';
 import { MongoCommand } from './MongoCommand';
 import { MongoDatabaseTreeItem } from './tree/MongoDatabaseTreeItem';
 
+const output = vscodeUtil.getOutputChannel();
+const notInScrapbookMessage = "This command can only be run inside of a MongoDB scrapbook (*.mongo)";
+
 export class MongoCommands {
+
+	public static async executeAllCommandsFromActiveEditor(database: IAzureParentNode<MongoDatabaseTreeItem>, extensionPath, editorManager: CosmosEditorManager, tree: AzureTreeDataProvider, context: IActionContext): Promise<void> {
+		output.show();
+		output.appendLine("Running all commands in scrapbook...")
+		let commands = MongoCommands.getAllCommandsFromActiveEditor();
+		await MongoCommands.executeCommands(commands, vscode.window.activeTextEditor, database, extensionPath, editorManager, tree, context);
+	}
+
 	public static async executeCommandFromActiveEditor(database: IAzureParentNode<MongoDatabaseTreeItem>, extensionPath, editorManager: CosmosEditorManager, tree: AzureTreeDataProvider, context: IActionContext): Promise<void> {
+		const commands = MongoCommands.getAllCommandsFromActiveEditor();
 		const activeEditor = vscode.window.activeTextEditor;
-		if (activeEditor.document.languageId !== 'mongo') {
-			vscode.window.showInformationMessage("This command can only be run inside of a MongoDB scrapbook (*.mongo)");
-			return;
-		}
 		const selection = activeEditor.selection;
-		const command = MongoCommands.getCommand(activeEditor.document.getText(), selection.start);
+		const command = MongoCommands.findCommandAtPosition(commands, selection.start);
+		return this.executeCommand(activeEditor, database, extensionPath, editorManager, tree, context, command);
```

Comment:
1. The order of these arguments is slightly different from the executeCommands function. Would it make sense to make these consistent? 
2. Why is this returned when the other use case (inside executeCommands) awaited? 

As an aside, would it make sense to just have one function, `executeCommands`, with this specific case having a function call - `await MongoCommands.executeCommands([command], vscode.window.activeTextEditor, database, extensionPath, editorManager, tree, context)` where we pass the command in an array? I think there might be lesser code, but it could seem less readable.

Reply:
I prefer the readability